### ISSUE TITLE
Phasing with whatshap phased variants

### ISSUE THREAD

**Issue by weishwu** [odelaneau/shapeit4#17]
I've phased variants from about 600 samples using whatshap with a pedigree file. About 100 samples in this file are in trios. I'm trying to use Shapeit4 to do statistical phasing for the remaining 500 samples that are not in trios (i.e. the samples whose parents were not sequenced). Since all these 600 samples are from a small isolated village so I wonder if the haplotypes resolved from the 100 samples could improve the phasing of the non-trio samples via statistical phasing. How should I set this up with Shapeit4? Could I just use the whathap phased vcf file as the input for Shapeit4 and Shapeit4 will use the phasing information automatically? Or should I prepare a scaffold vcf file using the phased variants from the 100 trio-samples and provide it to Shapeit4 using the "--scaffold" parameter?
Thanks.

**Comment by odelaneau** [comment]
Hi,

I'd suggest to phase all samples together (trios or not) using the partially resolved haplotypes as a scaffold (see documentation). For the scaffold, you need an additional vcf/bcf file (specified with --scaffold) with phased hets as 0|1 or 1|0 and unphased variants (e.g. triple hets) as 0/1.
ps: please use the last version, or you might have an "underflow" error.

Best,

**Comment by weishwu** [comment]
Thanks. Then the scaffold vcf (containing the 100 trio samples) will be a subset of the main input vcf (containing the 100 trio samples and the 500 non-trio samples). Is this OK?

**Comment by odelaneau** [comment]
Yes. But you need to make sure that only hets phased thanks to trio info are specified as phased (i.e. "|"), not those phased using sequencing reads.

**Comment by weishwu** [comment]
When I phased the variants using whatshap, I provided both pedigree and BAMs for whatshap to do both Mendelian phasing and read-based phasing. Do I have to redo whatshap phasing using only pedigree? Why can't the read-based phasing be included?

**Comment by odelaneau** [comment]
FYI, shapeit4 will use as scaffolded genotypes all those that are specified as phased (using "|"). This file therefore need to contain only genotypes that have been scaffolded thanks to family information. Somehow these guys are all in a single fixed phase set.

The main genotype file may contain  phase sets; so genotypes that are phased relative to other ones. In this case, shapeit4 may correct some of this phasing information when it is too discordant with what you get from statistical (or LD based) phasing.

Unfortunately, shapeit4 can not read both layers of information from the same unique file.  Hope this makes sense, and happy to give more details.

**Comment by weishwu** [comment]
Thanks for the explanation!

**Comment by weishwu** [comment]
I used only pedigree-based phased trios for the --scaffold parameter, but for the main input vcf, I tried two files. Both files contain the variants of one trio, and the difference is that the variants are unphased in file A, while file B contains the same variants but they were phased by whatshap with BAM but without pedigree (so they were read-based phased). I thought read-based phasing would help Shapeit4 phase more variants, however file A turned out to produce more phased variants. I also looked the ratio of the phased variants that agreed with a pedigree-based phasing of these variants, and it turned out file A (unphased) had a slightly better validation rate (90%) than file B (pre-phased by whatshap based on reads; 87%). Why does this happen? Does this mean I should not do read-based pre-phasing before running Shapeit4?

**Comment by weishwu** [comment]
The variants used by --scaffold were phased by whatshap with a pedigree file and they always have paternal alleles in front of maternal alleles (P|M). I ran Shapeit4 with this scaffold and got all samples with no trios phased. Do they phasing results still have this P|M order? If not, how can I combine the phased variants between different chromosomes?
